File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letter = 0;
    int word = 1;
    int sentence = 0;
    //check what is inside the string
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        char character = text[i];
        if (isalpha(character))
        {
            letter ++;
        }
        else if (character == ' ')
        {
            word ++;
        }
        else if (character == '!' || character == '?' || character == '.')
        {
            sentence ++;
        }
    }
    //grade calculation
    int grade = round(0.0588 * (float) letter / (float) word * 100 - 0.296 * (float) sentence / (float) word * 100 - 15.8);
    if (1 <= grade && grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string
from math import *

text =  get_string("Text: ")
length = len(text)
letter = 0
word = 0
sentence = 0
for i in range(length):
    char = text[i]
    if char == ' ':
        word += 1
    elif char == '!' or char == '?' or char == '.':
        sentence += 1
    else:
        letter += 1
grade = round(0.0588 * letter / word * 100 - 0.296 * sentence / word * 100 - 15.8)

if 1 < grade < 16:
    print(f"Grade {grade}")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade 16+")